Version 2.0.0 Release

avantonder opened the pull request:
<!--
# nf-core/bactmap pull request

Many thanks for contributing to nf-core/bactmap!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the dev branch, unless you're preparing a pipeline release.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/bactmap/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/bactmap/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/bactmap _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

Commit:
modules and subworkflows

avantonder committed:
Merge pull request #110 from avantonder/dev

modules and subworkflows

sainsachiko committed:
add snp-sites

sainsachiko committed:
add filtlong

sainsachiko committed:
add nanoq

sainsachiko committed:
add porechop

sainsachiko committed:
add porechop/abi

sainsachiko committed:
Configure added modules

avantonder committed:
Merge pull request #116 from sainsachiko/add-modules

Add modules

MareikeJaniak committed:
modules : add bctools filter module

MareikeJaniak committed:
reverse manual bcftools filter addition

MareikeJaniak committed:
adding bcftools filter module from nf-core modules

project-defiant committed:
feat: samplesheet schema update

sgsutcliffe committed:
Add RASUSA module

avantonder committed:
Merge pull request #117 from MareikeJaniak/dev

add module bcftools filter

avantonder committed:
Merge pull request #118 from sgsutcliffe/dev

Add RASUSA module

MareikeJaniak committed:
add cat module from nf-core modules

project-defiant committed:
feat: add samplesheet

project-defiant committed:
test: test and full test config

sgsutcliffe committed:
Add CLAIR3 module

sgsutcliffe committed:
Merge pull request #1 from nf-core/dev

project-defiant committed:
feat: add reference parameter

project-defiant committed:
feat: make sure new samplesheet params are required

project-defiant committed:
feat: test config

project-defiant committed:
Merge pull request #122 from sgsutcliffe/dev

Add CLAIR3 module

project-defiant committed:
feat: ensure reference param exists in nextflow-schema.json

project-defiant committed:
chore: apply nf-core linting

opetryk committed:
initial commit

opetryk committed:
merge dev

avantonder committed:
Merge branch 'dev' into dev

avantonder committed:
Merge pull request #120 from MareikeJaniak/dev

add cat module from nf-core modules

avantonder committed:
Merge pull request #121 from project-defiant/104-nf-core-test-2

feat: samplesheet schema update

avantonder committed:
Merge pull request #123 from project-defiant/linting

Pipeline linting

Commit:
avt update 250325

opetryk committed:
Add minimap2 to the pipeline

opetryk committed:
Merge pull request #1 from opetryk/minimap2

Minimap2

opetryk committed:
Merge upstream origin dev to lcal dev

avantonder committed:
Update workflows/bactmap.nf

Co-authored-by: Szymon Szyszkowski <69353402+project-defiant@users.noreply.github.com>

avantonder committed:
Update workflows/bactmap.nf

Co-authored-by: Szymon Szyszkowski <69353402+project-defiant@users.noreply.github.com>

avantonder committed:
Merge pull request #125 from avantonder/dev

avt update 250325

avantonder committed:
Merge pull request #126 from opetryk/dev

Add minimap2 to the pipeline

project-defiant committed:
chore: formatting typos

avantonder committed:
Merge pull request #130 from project-defiant/szsz-formatting-typos

chore: formatting typos

Commit:
add fastp & bcftools stats

avantonder committed:
Merge pull request #131 from avantonder/dev

add fastp & bcftools stats

project-defiant committed:
chore: remove redundand path segment

project-defiant committed:
chore: remove redundand path segment

avantonder committed:
Merge pull request #132 from project-defiant/szsz-fix-test-dataset-path

chore: remove redundand path segment

QuadjoLegend committed:
Add bwamem2/mem

QuadjoLegend committed:
Add files via upload

Add bwamem2/mem module

Commit:
working to test

project-defiant committed:
Merge pull request #134 from avantonder/dev

working to test

QuadjoLegend committed:
Update and rename modules/nf-core/bwamem2/mem/git to modules/nf-core/bwamem2/mem/tests/tags.yml

QuadjoLegend committed:
Add files via upload

QuadjoLegend committed:
Create tags.yml

QuadjoLegend committed:
Create main.nf.test

QuadjoLegend committed:
Create main.nf.test.snap

QuadjoLegend committed:
Create environment.yml

QuadjoLegend committed:
Create main.nf

QuadjoLegend committed:
Create meta.yml

QuadjoLegend committed:
Create tags.yml

QuadjoLegend committed:
Create main.nf.test

QuadjoLegend committed:
Create main.nf.test.snap

QuadjoLegend committed:
Create environment.yml

QuadjoLegend committed:
Create main.nf

QuadjoLegend committed:
Create meta.yml

MareikeJaniak committed:
adding modules for consensus_bcftools workflow

avantonder committed:
Merge branch 'dev' into dev

avantonder committed:
Merge pull request #135 from QuadjoLegend/dev

Add bwamem2/mem fastqscan and falco modules

avantonder committed:
Merge pull request #136 from MareikeJaniak/dev

adding modules for consensus_bcftools workflow

sgsutcliffe committed:
Added the prototype for a subworkflow consensus_bcftools

sgsutcliffe committed:
Added the prototype for a subworkflow consensus_bcftools

sgsutcliffe committed:
Merge branch 'dev' of https://github.com/sgsutcliffe/bactmap into dev

sgsutcliffe committed:
Accidently commited my codespace devcontainer

IsmaelHC1994 committed:
hard-coded for bcftools_consensus subworkflow. mainly the problem was the awk pipe at the end, we decided to go with a local version of genomecov.
this is a simpler version than the nf-core version, it expects a bam as interval file, and it does not expect sorting either.
we did add a threshold parameter in the nextflow.config in case the user wants to specify the threshold for low coverage

project-defiant committed:
fix: ensure the logic of ONT parsing is correct

project-defiant committed:
chore: drop redundant null defaults

avantonder committed:
Merge pull request #139 from project-defiant/empty-array-in-fastq_2

fix: ensure the logic of ONT parsing is correct

Commit:
first pipeline run

sgsutcliffe committed:
Add single process label

sgsutcliffe committed:
nf-core formatted channels & remove commented code

avantonder committed:
Merge pull request #140 from avantonder/dev

first pipeline run

avantonder committed:
Merge pull request #141 from IsmaelHC1994/dev

hard-coded for bcftools_consensus subworkflow. mainly the problem was…

avantonder committed:
Merge branch 'dev' into dev

avantonder committed:
Merge pull request #138 from sgsutcliffe/dev

Add the consensus_bcftools subworkflow prototype

Commit:
Longread mapping creation

Commit:
Update schema

Commit:
Update schema

sainsachiko committed:
hot-fix: add missing closure to modules.config

sainsachiko committed:
add module seqtk_comp

QuadjoLegend committed:
Update multiqc_config.yml

avantonder committed:
Merge pull request #144 from nf-core/QuadjoLegend-patch-2

Update multiqc_config.yml

avantonder committed:
Merge pull request #143 from sainsachiko/dev

hot-fix: add missing closure to modules.config

avantonder committed:
Merge branch 'dev' into longest-mapping

avantonder committed:
Merge pull request #142 from opetryk/longest-mapping

Longread mapping creation

Commit:
avt update 270325

avantonder committed:
Merge pull request #145 from avantonder/dev

avt update 270325

sainsachiko committed:
remove error-causing comma

sainsachiko committed:
add mapping_statistics subworkflow

sainsachiko committed:
Merge branch 'nf-core:dev' into dev

avantonder committed:
Merge pull request #146 from sainsachiko/dev

Add subworkflow mapping statistics

Commit:
first working version

avantonder committed:
Update CHANGELOG.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update CHANGELOG.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update CHANGELOG.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update CHANGELOG.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update README.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update README.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update workflows/bactmap.nf

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update subworkflows/local/consensus_bcftools/main.nf

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update README.md

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
Update bin/calculate_fraction_of_non_GATC_bases.py

Co-authored-by: Evangelos Karatzas <32259775+vagkaratzas@users.noreply.github.com>

avantonder committed:
avt update 050825

avantonder committed:
avt update 050825

avantonder committed:
modules update

avantonder committed:
modules update

avantonder committed:
modules update

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix linting issues

avantonder committed:
fix multiqc config

avantonder committed:
fix multiqc config

avantonder committed:
fix linting issues

avantonder committed:
ready for review

avantonder committed:
ready for review

avantonder committed:
ready for review

avantonder committed:
ready for review

avantonder committed:
ready for review

avantonder committed:
ready for review

avantonder committed:
Merge pull request #151 from avantonder/dev

Pre-release dev branch

avantonder committed:
pre-release dev

avantonder committed:
Merge pull request #152 from avantonder/dev

pre-release dev

Joon-Klaps commented on .github/workflows/ci.yml:
Think this file can be deleted right? The default test of `nf-test` also uses the `test` profile. So testing it with the old CI seems redundant now.

avantonder replied:
Deleted

Joon-Klaps commented on .github/workflows/awsfulltest.yml:
```diff
@@ -4,64 +4,40 @@ name: nf-core AWS full size tests
             {
               "hook_url": "${{ secrets.MEGATESTS_ALERTS_SLACK_HOOK_URL }}",
-              "outdir": "s3://${{ secrets.AWS_S3_BUCKET }}/bactmap/results-${{ github.sha }}"
+              "outdir": "s3://${{ secrets.AWS_S3_BUCKET }}/oncoanalyser/results-${{ steps.revision.outputs.revision }}"
```
This seems like a copy & paste error?

avantonder replied:
Yes, indeed. Fixed

Joon-Klaps commented on bin/fastqscan_parser.py:
```diff
@@ -0,0 +1,68 @@
+    make_dir(out_dir)
+
+    ## Create list of fastq-scan json files
+    json_files = sorted(glob.glob('*.json'))
```
nitpicking 
I think it would be better if you were to ask for an input dir in args and search there for json files instead of working dir.

Joon-Klaps commented on bin/multi2single_sequence.py:
```diff
@@ -0,0 +1,58 @@
+    return parser
+
+def combine_sequences(reference_sequence):
+    records = list(SeqIO.parse(reference_sequence, "fasta"))
+    new_sequence = ''.join([str(record.seq) for record in records])
+    new_id = '|'.join([record.id for record in records])
```
nitpicking
If the Multifasta is really, really big, I thought that one might encounter OOM by converting it to a list which reads it all into mem. If you are confident that this will never occur, ignore this. If so maybe something like this? Haven't tested this
```suggestion
    with open("merged.fasta", "w") as f:
    f.write(">")  # header placeholder

    for record in SeqIO.parse(reference_sequence, "fasta"):
        ids.append(record.id)
        f.write(str(record.seq))

    # rewrite header with all IDs at the start
    new_id = '|'.join(ids)
    if len(new_id) > 100:
        new_id = new_id[:97] + '...'
    f.seek(0)
    f.write(new_id + "\n")
 ```

avantonder replied:
The pipeline is designed for mapping sequence data to bacterial reference sequences. The genomes are never going to be more than 10Mb.

Joon-Klaps commented on bin/read_stats.py:
```diff
@@ -0,0 +1,85 @@
+    make_dir(out_dir)
+
+    ## Create list of raw reads fastq-scan json files
+    raw_json_files = sorted(glob.glob('*.raw.json'))
+
+    ## Create list of processed reads fastq-scan json files
+    processed_json_files = sorted(glob.glob('*.processed.json'))
```
nitpicking... again sorry
Same here, think it would be better if you were to ask for input params & use in nextflow the stageAs.

Joon-Klaps commented on bin/read_stats_parser.py:
```diff
@@ -0,0 +1,58 @@
+    make_dir(out_dir)
+
+    ## Create list of cluster files
+    read_stats_files = sorted(glob.glob('*.csv'))
```
🤷‍♂️

Joon-Klaps commented on bin/seqtk_parser.py:
```diff
@@ -0,0 +1,72 @@
+    make_dir(out_dir)
+
+    ## Create list of seqtk comp tsv outputs
+    seqtk_files = sorted(glob.glob("*.tsv"))
```
🤷‍♂️

Joon-Klaps commented on docs/images/bactmap_pipeline.png:
Think you missed to connect your blue ref line with minimap, as there seems to be an unconnected blue line at the bottom
<img width="1133" height="452" alt="image" src="https://github.com/user-attachments/assets/0dee741f-5bc3-4ab5-954c-bb17ad599717" />

Joon-Klaps commented on modules/local/clair3/tests/main.nf.test:
```diff
@@ -0,0 +1,104 @@
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                                process.out.vcf.collect { file(it[1]).getName() },
+                                process.out.tbi.collect { file(it[1]).getName() },
+                                process.out.versions,
+                                process.out.phased_vcf.collect { file(it[1]).getName() },
+                                process.out.phased_tbi.collect { file(it[1]).getName() }).match()}
+            )
```
I'm guessing it's not stable due to the vcf header? 
Might be interesting to use `nft-vcf` plugin with it's function `.variantsMD5`

avantonder replied:
I'm not sure how to address this. This was the original main.nf.test from nf-core.

Joon-Klaps commented on modules/local/clair3/tests/main.nf.test:
```diff
@@ -0,0 +1,104 @@
+                                process.out.vcf.collect { file(it[1]).getName() },
+                                process.out.tbi.collect { file(it[1]).getName() },
+                                process.out.versions,
+                                process.out.phased_vcf.collect { file(it[1]).getName() },
+                                process.out.phased_tbi.collect { file(it[1]).getName() }).match()}
```
phased_vcf & phased_tbi their names are not stored in snapshot. It's an empty array.

avantonder replied:
I'm not sure how to address this. This was the original main.nf.test from nf-core.

Joon-Klaps commented on modules/local/rasusa/main.nf:
Think this module would also be a nice addition to nf-core. Something to keep in mind for future releases.

avantonder replied:
It is an nf-core module. I moved it to local as the updated module had changed the format of the input channels so I reverted to the old version as it was simpler.

Joon-Klaps commented on subworkflows/local/minimap2_alignment/main.nf:
Something to keep in mind as i see this coming up quite. few times across different subworkflows, that according to nf-core rules, it's good pratice not to use params in the workflow, pass them as input arguments. A

But I also believe it to be cleaner to not pass them, despite it being more challenging to understand the entire workflow.

avantonder replied:
Ok, I will bear this in mind

Joon-Klaps commented on workflows/bactmap.nf:
```diff
@@ -76,6 +404,31 @@ workflow BACTMAP {
+    ch_multiqc_files = ch_multiqc_files.mix(SHORTREAD_MAPPING.out.mqc.collect{it[1]}.ifEmpty([]))
+
+    ch_multiqc_files = ch_multiqc_files.mix(LONGREAD_MAPPING.out.mqc.collect{it[1]}.ifEmpty([]))
+
```
Why not declare this after the corresponding subworkflows are run? It would be a lot easier to maintain.

avantonder replied:
I mainly follow out nf-core workflows eg. taxprofiler when laying out my own. I guess I would argue that having all the multiqc inputs about the MULTIQC process makes it easier to see what's going in.

Joon-Klaps reviewed (commented):
Very nice work, no major comments. Only suggestions.

avantonder committed:
post pre-release review

avantonder committed:
Merge pull request #154 from avantonder/dev

post pre-release review

avantonder committed:
update template files

avantonder committed:
update template files

avantonder committed:
Merge pull request #156 from avantonder/dev

update template files

avantonder committed:
template updates

avantonder committed:
adjust mapping

avantonder committed:
pipeline fixes

avantonder committed:
pipeline fixes

avantonder committed:
pipeline fixes

avantonder committed:
pipeline fixes

avantonder committed:
pipeline fixes

avantonder committed:
pipeline fixes

avantonder committed:
update logos

avantonder committed:
Merge pull request #158 from avantonder/dev

Dev

nf-core-bot committed:
WIP: Remove vulnerable PR-comment artifact pattern

Automated security rollout could not apply this cleanly. Resolve the remaining conflict markers and *.rej files, then open a PR from 'patch' into 'dev'.

mashehu committed:
Merge pull request #162 from nf-core/patch

Remove vulnerable PR-comment artifact pattern
